sleap_io/io/nwb.py

```diff
@@ -6,27 +6,298 @@
 import datetime
 import uuid
 import re
+from imageio.v3 import imwrite
 
-import pandas as pd  # type: ignore[import]
+import pandas as pd
 import numpy as np
 
 try:
     from numpy.typing import ArrayLike
 except ImportError:
     ArrayLike = np.ndarray
-from pynwb import NWBFile, NWBHDF5IO, ProcessingModule  # type: ignore[import]
-from ndx_pose import PoseEstimationSeries, PoseEstimation  # type: ignore[import]
+
+from pynwb import NWBFile, NWBHDF5IO, ProcessingModule
+from pynwb.file import Subject
+from pynwb.image import ImageSeries
+from pynwb.testing.mock.utils import name_generator
+
+from ndx_pose import (
+    PoseEstimationSeries,
+    PoseEstimation,
+    Skeleton as NWBSkeleton,
+    Skeletons,
+    SkeletonInstance,
+    SkeletonInstances,
+    TrainingFrame,
+    TrainingFrames,
+    PoseTraining,
+    SourceVideos,
+)
 
 from sleap_io import (
     Labels,
     Video,
     LabeledFrame,
     Track,
-    Skeleton,
+    Skeleton as SLEAPSkeleton,
     Instance,
     PredictedInstance,
+    Edge,
+    Node,
 )
 from sleap_io.io.utils import convert_predictions_to_dataframe
+from sleap_io.io.main import load_slp, save_nwb
+
+
+def pose_training_to_labels(pose_training: PoseTraining) -> Labels:  # type: ignore[return]
+    """Creates a Labels object from an NWB PoseTraining object.
+
+    Args:
+        pose_training: An NWB PoseTraining object.
+
+    Returns:
+        A Labels object.
+    """
+    if not isinstance(pose_training, PoseTraining):
+        raise ValueError("The input must be an NWB PoseTraining object.")
+    
+    labeled_frames = []
+    for training_frame in pose_training.training_frames.training_frames.values():
+        video = Video(filename=f"{training_frame.source_video}")
+        frame_idx = training_frame.source_video_frame_index
+        instances = [
+            Instance.from_numpy(
```

Comment:
Is the `track` field not stored in the NWB `SkeletonInstance`? If not, let's note it and let Ryan know that we need an update.